snuba/cli/subscriptions_executor.py

```diff
@@ -23,15 +23,24 @@
     "--dataset",
     "dataset_name",
     required=True,
-    type=click.Choice(["events", "transactions", "metrics"]),
+    type=click.Choice(["events", "transactions", "metrics", "generic_metrics"]),
     help="The dataset to target.",
 )
 @click.option(
     "--entity",
     "entity_names",
     required=True,
     multiple=True,
-    type=click.Choice(["events", "transactions", "metrics_counters", "metrics_sets"]),
+    type=click.Choice(
+        [
+            "events",
+            "transactions",
+            "metrics_counters",
+            "metrics_sets",
+            "generic_metrics_sets",
+            "generic_metrics_distributions",
+        ]
```

Comment:
Following up on my comment from a previous PR: https://github.com/getsentry/snuba/pull/2950#discussion_r923849164

I didn't mean have the choice be all values from the `EntityKey` Enum, but to generate this list of choices using values from the Enum.

ie I'm suggesting we replace this list of strings with something like:
```python3
[
    EntityKey.EVENTS.value,
    ...,
    EntityKey.GENERIC_METRICS_DISTRIBUTIONS.value,
]
```
where the list _only_ contains values we want, not all possible entities. Although not the ideal non-code config solution, it's probably still better than hardcoding strings.

Reply:
sure, I think that makes sense